src/binary/non_blocking/type_descriptor.rs

```diff
@@ -9,7 +9,7 @@ use crate::IonType;
 /// [Typed Value Formats](https://amazon-ion.github.io/ion-docs/docs/binary.html#typed-value-formats)
 /// section of the binary Ion spec.
 #[derive(Copy, Clone, Debug, PartialEq)]
-pub(crate) struct TypeDescriptor {
+pub struct TypeDescriptor {
```

Comment:
🗺️ This PR makes a couple of types `pub` while leaving their module `pub(crate)` or even private. A later change in this diff re-exports these `pub` types if the `experimental-lazy-reader` feature is enabled, allowing the benchmarks to use some of these implementation details.